include/merlin/core_kernels.cuh

```diff
@@ -1025,6 +1247,66 @@ struct SelectUpsertKernelWithIO {
   }
 };
 
+template <typename K, typename V, typename M>
+struct SelectUpsertAndEvictKernelWithIO {
+  static void execute_kernel(
+      const float& load_factor, const int& block_size, cudaStream_t& stream,
+      const size_t& n, const int c_table_index_,
+      const Table<K, V, M>* __restrict table, const K* __restrict keys,
+      const V* __restrict values, const M* __restrict metas,
+      K* __restrict evicted_keys, V* __restrict evicted_values,
+      M* __restrict evicted_metas) {
+    if (load_factor <= 0.5) {
```

Comment:
Possibly, the kernel has different best-performance configuration from the `upsert_kernel_with_io`,  but is not a big deal, currently, just copy the configuration of `upsert_kernel_with_io` is OK.